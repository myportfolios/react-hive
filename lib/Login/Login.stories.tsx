import type { Meta, StoryObj } from "@storybook/react";
import { Login } from "./Login";

const ActionData = {};
const meta: Meta<typeof Login> = {
  title: "Components/Login",
  component: Login,
  tags: ["autodocs"],
  parameters: {
    layout: "centered",
  },
  excludeStories: /.*Data$/,
  args: {
    ...ActionData,
  },
};

export default meta;

export const Default: StoryObj = {
  args: {
    title: "Enter details to login",
  },
};
