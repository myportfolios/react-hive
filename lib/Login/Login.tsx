import React, { useState } from "react";
import { TextInput } from "../TextInput/TextInput";
import { Button } from "react-hive";

interface LoginProps {
  title?: string;
  type: string;
}
export function Login({ title }: LoginProps) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleInput = (value: string, type: string) => {
    switch (type) {
      case "email":
        setEmail(value);
        break;
      case "password":
        setPassword(value);
        break;
      default:
        return;
    }
  };
  return (
    <div className="flex flex-col w-64 items-center p-6 space-y-2 border border-primary shadow-lg">
      <span className="label">{title}</span>
      <TextInput
        handleUserInput={handleInput}
        type="email"
        value={email}
        isVertical
        label="Email"
      />
      <TextInput
        handleUserInput={handleInput}
        type="password"
        value={password}
        isVertical
        label="Password"
      />
      <Button>{"Submit"}</Button>
    </div>
  );
}

/**
 * To do
 * 1. Test login component - input and  submit button
 * 2. Create label for each input, set focus on label when component loads
 * 3. Test Error state for input
 */
