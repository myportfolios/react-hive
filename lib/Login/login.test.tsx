import { render } from "@testing-library/react";

import { Login } from "./Login";

describe(`should render ${Login.name}`, () => {
  it("should render", () => {
    const { container } = render(<Login />);

    expect(container).toMatchSnapshot();
  });
});
