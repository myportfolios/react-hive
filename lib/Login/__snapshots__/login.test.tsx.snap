// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Login > should render 1`] = `
<div>
  <div>
    Login
  </div>
</div>
`;
